### PR TITLE
Handling for reserved Python keywords as JsProxy attrs

### LINKED ISSUE #3615: `Array.from` in Python Causes Syntax Error
## 🐛 Bug

The [Array.from()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from) static method (alternative constructor) throws a syntax error when used in Python.

### To Reproduce

The following code throws a SyntaxError: 

```python
<script>
  async function main() {
    let pyodide = await loadPyodide();
    pyodide.runPython(`
      from js import Array
      a = Array.from([1,2,3])
      print(a)
      `);
  };
  main();
</script>
```
```
...
 File "/lib/python3.10/_pyodide/_base.py", line 144, in _parse_and_compile_gen
 mod = compile(source, filename, mode, flags | ast.PyCF_ONLY_AST)
  File "<exec>", line 3
    a = Array.from([1,2,3])
              ^^^^
SyntaxError: invalid syntax
```

However, the following code works:
```python
<script>
  async function main() {
    let pyodide = await loadPyodide();
    const generator = pyodide.runPython(`
      from js import Array
      fromFunc = getattr(Array, 'from')
      b = fromFunc([4,5,6])
      print(b)
      `);
  };
  main();
</script>
```

### Expected behavior

This behavior is, in a way, consistent with Python, where directly setting to or accessing the `from` attribute of an object is also a syntax error, but you can create one with `setattr` and `getattr` (or similar):

```python
class A():
        ...

#SyntaxError
A.from = lambda: print("Whoops") 

#Works
setattr(A, 'from', lambda : print("from() was called!"))
getattr(A, 'from')()
```

But it's unfortunate that the simple Array.from() method is made more cumbersome to use by the Python keyword. As for the solution? I don't know - could be something like detecting properties on JS objects that are Python keywords, and creating an additional property for them? (`_from`?). Or maybe just highlighting the issue of JS objects with properties that are Python keywords is enough.

Forgive me if that is already documented somewhere, I couldn't find a specific reference to it.

### Environment

- Pyodide Version 0.22.1
- Browser version Chrome Version 110.0.5481.178 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by hoodmane**
Resolves #3615. Things like `from` and `finally` are reserved keywords in Python and so `a.finally` 
is a `SyntaxError`. This automatically reroutes `a.from_` on a `JsProxy` to refer to `a.from` so it
can be used reasonably conveniently from Python.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

**Review by rth — APPROVED** [review]
Thanks @hoodmane !

(Needs a changelog)

**Comment by hoodmane** [comment]
Thanks for the review. Worth noting that one bad aspect of this is that:
```py
for x in dir(p):
  delattr(p,x)
```
will fail. I see not much reason to ever do that. To help with tab complete and to fix this very obscure detail, we may want to only include e.g. `from_` and not include `from` in `dir` so `o.fr<tab>` completes to `o.from_` which works instead of `o.from` which does not. This also keeps these attributes from getting double counted in `dir`.

File: src/core/hiwire.c [hunk 1]
```diff
@@ -908,16 +908,43 @@ EM_JS_REF(JsRef, JsObject_New, (), {
...
   let jsobj = Hiwire.get_value(idobj);
   let jskey = UTF8ToString(ptrkey);
   let result = jsobj[jskey];
   // clang-format off
-  if (result === undefined && !(jskey in jsobj)) {
-    // clang-format on
+  if (result !== undefined || jskey in jsobj) {
+    return Hiwire.new_value(result);
+  }
+  if(!jskey.endsWith("_")) {
```
Comment (ryanking13) [inline src/core/hiwire.c:932]: Some comments here about what we are doing wouldn't hurt.

**Review by ryanking13 — APPROVED** [review]
Thanks!

**Comment by hoodmane** [comment]
Okay I changed the logic of this since it was last reviewed so I would appreciate one more review.

File: src/core/hiwire.c [hunk 2]
```diff
@@ -908,33 +908,82 @@ EM_JS_REF(JsRef, JsObject_New, (), {
...
+ */
+EM_JS(int, normalizeReservedWords, (int action, int word), {
+  // clang-format off
+  // 1. if word is not a reserved word followed by 0 or more underscores, return
+  //    it unchanged.
+  const noTrailing_ = word.replace(/_*$/, "");
+  if (!isReservedWord(noTrailing_)) {
+    return word;
+  }
+  // 2. If there is at lease one trailing underscore, return the word with a
```
Comment (ryanking13) [inline src/core/hiwire.c:948]:
```suggestion
  // 2. If there is at least one trailing underscore, return the word with a
```

**Review by ryanking13 — APPROVED** [review]
Great work, thanks @hoodmane!

**Pull request merged**